In [1]:
import pandas as pd
import sqlite3
pd.options.display.float_format = '{:,.2f}'.format
begInv = pd.read_csv('BegInvFINAL12312016Sample.csv')

In [2]:
begInv.head(10)

InventoryId  Store          City  Brand  \
0   1_HARDERSFIELD_58      1  HARDERSFIELD     58   
1   1_HARDERSFIELD_60      1  HARDERSFIELD     60   
2   1_HARDERSFIELD_62      1  HARDERSFIELD     62   
3   1_HARDERSFIELD_63      1  HARDERSFIELD     63   
4   1_HARDERSFIELD_72      1  HARDERSFIELD     72   
5   1_HARDERSFIELD_75      1  HARDERSFIELD     75   
6   1_HARDERSFIELD_77      1  HARDERSFIELD     77   
7   1_HARDERSFIELD_79      1  HARDERSFIELD     79   
8  1_HARDERSFIELD_115      1  HARDERSFIELD    115   
9  1_HARDERSFIELD_120      1  HARDERSFIELD    120   

                   Description   Size  onHand  Price   startDate  
0  Gekkeikan Black & Gold Sake  750mL       8  12.99  2016-01-01  
1       Canadian Club 1858 VAP  750mL       7  10.99  2016-01-01  
2     Herradura Silver Tequila  750mL       6  36.99  2016-01-01  
3   Herradura Reposado Tequila  750mL       3  38.99  2016-01-01  
4         No. 3 London Dry Gin  750mL       6  34.99  2016-01-01  
5    Three Olives Tomato Vodka  750mL      18  14.99  2016-01-01  
6  Three Olives Espresso Vodka  750mL       7  14.99  2016-01-01  
7     Three Olives Loopy Vodka  750mL       2  14.99  2016-01-01  
8              Belvedere Vodka  Liter       5  27.99  2016-01-01  
9  Tarantula Azul Tequila Gift  750mL      11  13.99  2016-01-01

# Deliverable 1A

In [3]:
begInv = begInv.rename(columns = {'onHand' : '01_01_2016_Qty'})

In [4]:
begInv['InvCost01_01_2016'] = (begInv['Price'] * begInv['01_01_2016_Qty'])
begInv.head(3)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   
2  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  01_01_2016_Qty  Price   startDate  InvCost01_01_2016  
0  750mL               8  12.99  2016-01-01             103.92  
1  750mL               7  10.99  2016-01-01              76.93  
2  750mL               6  36.99  2016-01-01             221.94

In [5]:
begInv_Store = begInv[['Store','01_01_2016_Qty','InvCost01_01_2016']].groupby('Store').sum()
begInv_Store.sort_values(by='InvCost01_01_2016', ascending=False)
begInv_Store.head(10)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
1               49917         838,079.59
2               52925         840,452.20
3                8379         132,241.23
4               56819         786,488.68
5               17968         218,720.64
6               48512         739,355.88
7               48330         775,694.77
8               38803         554,182.37
9               52169         713,163.99
10              84728       1,350,761.19

# Deliverable 1B

In [6]:
begInv_Brand = begInv[['Brand','Description','01_01_2016_Qty','InvCost01_01_2016']].groupby(['Brand','Description']).sum()
begInv_Brand.head(2)

01_01_2016_Qty  InvCost01_01_2016
Brand Description                                                   
58    Gekkeikan Black & Gold Sake             281           3,650.19
60    Canadian Club 1858 VAP                  288           3,165.12

In [7]:
begInv_Brand.sort_values(by='InvCost01_01_2016', ascending=False)
begInv_Brand.head(10)

01_01_2016_Qty  InvCost01_01_2016
Brand Description                                                    
58    Gekkeikan Black & Gold Sake              281           3,650.19
60    Canadian Club 1858 VAP                   288           3,165.12
61    Margaritaville Silver                     12             167.88
62    Herradura Silver Tequila                 384          14,204.16
63    Herradura Reposado Tequila               365          14,231.35
72    No. 3 London Dry Gin                     151           5,283.49
75    Three Olives Tomato Vodka                 18             269.82
77    Three Olives Espresso Vodka             1261          18,902.39
79    Three Olives Loopy Vodka                 902          13,520.98
82    Hundred Acre Ark Vy Cab Svgn              24           8,999.76

# Deliverable 2A

In [8]:
endInv = pd.read_csv('EndInvFINAL12312016.csv')
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  onHand  Price     endDate  
0  750mL      11  12.99  2016-12-31  
1  750mL       7  36.99  2016-12-31

In [9]:
endInv = endInv.rename(columns = {'onHand' : '12_31_2016_Qty'})
endInv.head()

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   
2  1_HARDERSFIELD_63      1  HARDERSFIELD     63   Herradura Reposado Tequila   
3  1_HARDERSFIELD_72      1  HARDERSFIELD     72         No. 3 London Dry Gin   
4  1_HARDERSFIELD_75      1  HARDERSFIELD     75    Three Olives Tomato Vodka   

    Size  12_31_2016_Qty  Price     endDate  
0  750mL              11  12.99  2016-12-31  
1  750mL               7  36.99  2016-12-31  
2  750mL               7  38.99  2016-12-31  
3  750mL               4  34.99  2016-12-31  
4  750mL               7  14.99  2016-12-31

In [10]:
endInv['InvCost12_31_2016'] = (endInv['Price'] * endInv['12_31_2016_Qty'])
endInv.head(3)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   
2  1_HARDERSFIELD_63      1  HARDERSFIELD     63   Herradura Reposado Tequila   

    Size  12_31_2016_Qty  Price     endDate  InvCost12_31_2016  
0  750mL              11  12.99  2016-12-31             142.89  
1  750mL               7  36.99  2016-12-31             258.93  
2  750mL               7  38.99  2016-12-31             272.93

In [11]:
endInv_Store = endInv[['Store','12_31_2016_Qty','InvCost12_31_2016']].groupby('Store').sum().sort_values(by='InvCost12_31_2016', ascending=False)
endInv_Store.head(10)

12_31_2016_Qty  InvCost12_31_2016
Store                                   
50             260717       4,887,260.68
73             164589       3,254,662.81
67             163765       3,076,114.82
34             145829       3,074,616.75
76             143866       2,975,945.18
69             150848       2,968,678.82
66             144579       2,860,504.99
74             166015       2,803,645.13
38             129397       2,463,906.85
55             125584       2,234,836.35

# Deliverable 2B

In [12]:
endInv_Brand = endInv[['Brand','Description','12_31_2016_Qty','InvCost12_31_2016']].groupby(['Brand','Description']).sum()
endInv_Brand.head(2)

12_31_2016_Qty  InvCost12_31_2016
Brand Description                                                   
58    Gekkeikan Black & Gold Sake             385           5,001.15
60    Canadian Club 1858 VAP                  146           1,604.54

In [13]:
endInv_Brand.sort_values(by='InvCost12_31_2016', ascending=False)
endInv_Brand.head(10)

12_31_2016_Qty  InvCost12_31_2016
Brand Description                                                   
58    Gekkeikan Black & Gold Sake             385           5,001.15
60    Canadian Club 1858 VAP                  146           1,604.54
61    Margaritaville Silver                    12             167.88
62    Herradura Silver Tequila                459          16,978.41
63    Herradura Reposado Tequila              442          17,233.58
72    No. 3 London Dry Gin                    139           4,863.61
75    Three Olives Tomato Vodka                 7             104.93
77    Three Olives Espresso Vodka            1764          26,442.36
79    Three Olives Loopy Vodka                995          14,915.05
86    Don Julio Real Tequila                   18           5,399.82

# Deliverable 3

In [14]:
merge3A = pd.merge(left=begInv, right=endInv, how='outer', on=['InventoryId'])
merge3 = merge3A[['InventoryId','01_01_2016_Qty','InvCost01_01_2016','12_31_2016_Qty','InvCost12_31_2016']]
merge3 = merge3.fillna(0)
merge3.head()

InventoryId  01_01_2016_Qty  InvCost01_01_2016  12_31_2016_Qty  \
0  1_HARDERSFIELD_58            8.00             103.92           11.00   
1  1_HARDERSFIELD_60            7.00              76.93            0.00   
2  1_HARDERSFIELD_62            6.00             221.94            7.00   
3  1_HARDERSFIELD_63            3.00             116.97            7.00   
4  1_HARDERSFIELD_72            6.00             209.94            4.00   

   InvCost12_31_2016  
0             142.89  
1               0.00  
2             258.93  
3             272.93  
4             139.96

# Deliverable 4A

In [15]:
purchases = pd.read_csv('InvoicePurchases12-31-16Sample.csv')
purchases.head(2)

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   

       PODate     PayDate  Quantity  Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6   214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15   140.55     8.57      NaN

In [16]:
purchases4A = purchases[['VendorNumber','VendorName','Dollars']].groupby(['VendorNumber','VendorName']).sum().sort_values(by='Dollars',ascending=False)
purchases4A.head(10)

Dollars
VendorNumber VendorName                               
3960         DIAGEO NORTH AMERICA INC    50,959,796.85
4425         MARTIGNETTI COMPANIES       27,821,473.91
12546        JIM BEAM BRANDS COMPANY     24,203,151.05
17035        PERNOD RICARD USA           24,124,091.56
480          BACARDI USA INC             17,624,378.72
1392         CONSTELLATION BRANDS INC    15,573,917.90
1128         BROWN-FORMAN CORP           13,529,433.08
9165         ULTRA BEVERAGE COMPANY LLP  13,210,613.93
3252         E & J GALLO WINERY          12,289,608.09
9552         M S WALKER INC              10,935,817.30

# Deliverable 4B

In [17]:
purchases4B = purchases[['VendorNumber','VendorName','Freight']].groupby(['VendorNumber','VendorName']).sum().sort_values(by='Freight',ascending=False)
purchases4B.head(10)

Freight
VendorNumber VendorName                            
3960         DIAGEO NORTH AMERICA INC    257,032.07
4425         MARTIGNETTI COMPANIES       144,719.92
12546        JIM BEAM BRANDS COMPANY     123,880.97
17035        PERNOD RICARD USA           123,780.22
480          BACARDI USA INC              89,286.27
1392         CONSTELLATION BRANDS INC     79,528.99
1128         BROWN-FORMAN CORP            68,601.68
9165         ULTRA BEVERAGE COMPANY LLP   68,054.70
3252         E & J GALLO WINERY           61,966.91
9552         M S WALKER INC               55,551.82

# Deliverable 4C

In [18]:
purchases4C = purchases[['VendorNumber','VendorName','Dollars','Freight']].groupby(['VendorNumber','VendorName']).sum()
purchases4C = purchases4C[purchases4C["Dollars"]>=250000]
purchases4C.head()

Dollars   Freight
VendorNumber VendorName                                         
480          BACARDI USA INC             17,624,378.72 89,286.27
516          BANFI PRODUCTS CORP          1,628,866.68  8,510.41
653          STATE WINE & SPIRITS         1,529,682.04  8,014.98
660          SAZERAC NORTH AMERICA INC.   3,537,977.55 17,932.33
1128         BROWN-FORMAN CORP           13,529,433.08 68,601.68

In [19]:
purchases4C["Freight_per_Dollar%"] = ((purchases4C['Freight']/purchases4C['Dollars'])*100)
purchases4C = purchases4C.sort_values(by='Freight_per_Dollar%',ascending=False)
purchases4C.head(10)

Dollars    Freight  \
VendorNumber VendorName                                             
9625         WESTERN SPIRITS BEVERAGE CO    361,249.21   1,933.19   
1590         DIAGEO CHATEAU ESTATE WINES  1,365,472.83   7,259.75   
9744         FREDERICK WILDMAN & SONS       759,449.24   3,999.93   
653          STATE WINE & SPIRITS         1,529,682.04   8,014.98   
17031        FLAG HILL WINERY & VINEYARD    300,403.20   1,573.31   
516          BANFI PRODUCTS CORP          1,628,866.68   8,510.41   
4425         MARTIGNETTI COMPANIES       27,821,473.91 144,719.92   
8673         STE MICHELLE WINE ESTATES    3,086,650.70  15,919.70   
9815         WINE GROUP INC               5,258,636.79  27,100.41   
9165         ULTRA BEVERAGE COMPANY LLP  13,210,613.93  68,054.70   

                                          Freight_per_Dollar%  
VendorNumber VendorName                                        
9625         WESTERN SPIRITS BEVERAGE CO                 0.54  
1590         DIAGEO CHATEAU ESTATE WINES                 0.53  
9744         FREDERICK WILDMAN & SONS                    0.53  
653          STATE WINE & SPIRITS                        0.52  
17031        FLAG HILL WINERY & VINEYARD                 0.52  
516          BANFI PRODUCTS CORP                         0.52  
4425         MARTIGNETTI COMPANIES                       0.52  
8673         STE MICHELLE WINE ESTATES                   0.52  
9815         WINE GROUP INC                              0.52  
9165         ULTRA BEVERAGE COMPANY LLP                  0.52

In [20]:
purchases4C.tail(5)

Dollars   Freight  \
VendorNumber VendorName                                           
8352         LUXCO INC                   2,051,436.01 10,261.60   
3924         HEAVEN HILL DISTILLERIES    2,816,661.94 14,069.87   
3089         SIDNEY FRANK IMPORTING CO   1,715,908.88  8,549.55   
9819         TREASURY WINE ESTATES       2,978,686.40 14,836.57   
6359         OLE SMOKY DISTILLERY LLC      387,622.69  1,922.00   

                                          Freight_per_Dollar%  
VendorNumber VendorName                                        
8352         LUXCO INC                                   0.50  
3924         HEAVEN HILL DISTILLERIES                    0.50  
3089         SIDNEY FRANK IMPORTING CO                   0.50  
9819         TREASURY WINE ESTATES                       0.50  
6359         OLE SMOKY DISTILLERY LLC                    0.50

# Deliverable 4D

In [21]:
purchases = purchases[purchases["Freight"]>=100]
purchases = purchases[purchases["Quantity"]<=1000]
purchases4D = purchases[['VendorNumber','VendorName','Freight','Dollars','Quantity']].groupby(['VendorNumber','VendorName']).sum().sort_values(by='Freight',ascending=False)
purchases4D.head()

Freight    Dollars  Quantity
VendorNumber VendorName                                               
2561         EDRINGTON AMERICAS          2,944.67 570,627.52     18120
8664         STOLI GROUP,(USA) LLC         349.02  14,299.98       975
7239         REMY COINTREAU USA INC        348.07   7,515.38       382
653          STATE WINE & SPIRITS          303.70   5,544.89       541
6785         PALM BAY INTERNATIONAL INC    242.73   5,533.18       715

# Deliveralbe 4E

In [22]:
purchases4D["Freight_per_Dollar%"] = ((purchases4D['Freight']/purchases4D['Dollars'])*100)
purchases4D["Freight_per_Quantity"] = (purchases4D['Freight']/purchases4D['Quantity'])
purchases4E = purchases4D.sort_values(by='Freight_per_Dollar%',ascending=False)
purchases4E

Freight    Dollars  Quantity  \
VendorNumber VendorName                                                  
653          STATE WINE & SPIRITS          303.70   5,544.89       541   
7239         REMY COINTREAU USA INC        348.07   7,515.38       382   
6785         PALM BAY INTERNATIONAL INC    242.73   5,533.18       715   
2555         DISARONNO INTERNATIONAL LLC   146.86   3,506.41       385   
6213         MHW LTD                       177.50   4,821.83       412   
4692         KOBRAND CORPORATION           185.93   5,336.48       427   
1485         CASTLE BRANDS CORP.           179.26   5,420.41       320   
8352         LUXCO INC                     196.61   6,139.47       819   
10000        MAJESTIC FINE WINES           218.05   7,234.88       875   
3089         SIDNEY FRANK IMPORTING CO     133.83   4,443.66       379   
3924         HEAVEN HILL DISTILLERIES      200.02   7,079.02       818   
8664         STOLI GROUP,(USA) LLC         349.02  14,299.98       975   
9625         WESTERN SPIRITS BEVERAGE CO   169.46   8,657.44       795   
2242         DELICATO VINEYARDS INC        127.05   6,646.46       808   
1273         CALEDONIA SPIRITS INC         109.80  20,334.21       576   
4848         LAIRD & CO                    100.37  18,937.82       982   
2561         EDRINGTON AMERICAS          2,944.67 570,627.52     18120   
98450        Serralles Usa LLC             122.24  27,163.64       733   

                                          Freight_per_Dollar%  \
VendorNumber VendorName                                         
653          STATE WINE & SPIRITS                        5.48   
7239         REMY COINTREAU USA INC                      4.63   
6785         PALM BAY INTERNATIONAL INC                  4.39   
2555         DISARONNO INTERNATIONAL LLC                 4.19   
6213         MHW LTD                                     3.68   
4692         KOBRAND CORPORATION                         3.48   
1485         CASTLE BRANDS CORP.                         3.31   
8352         LUXCO INC                                   3.20   
10000        MAJESTIC FINE WINES                         3.01   
3089         SIDNEY FRANK IMPORTING CO                   3.01   
3924         HEAVEN HILL DISTILLERIES                    2.83   
8664         STOLI GROUP,(USA) LLC                       2.44   
9625         WESTERN SPIRITS BEVERAGE CO                 1.96   
2242         DELICATO VINEYARDS INC                      1.91   
1273         CALEDONIA SPIRITS INC                       0.54   
4848         LAIRD & CO                                  0.53   
2561         EDRINGTON AMERICAS                          0.52   
98450        Serralles Usa LLC                           0.45   

                                          Freight_per_Quantity  
VendorNumber VendorName                                         
653          STATE WINE & SPIRITS                         0.56  
7239         REMY COINTREAU USA INC                       0.91  
6785         PALM BAY INTERNATIONAL INC                   0.34  
2555         DISARONNO INTERNATIONAL LLC                  0.38  
6213         MHW LTD                                      0.43  
4692         KOBRAND CORPORATION                          0.44  
1485         CASTLE BRANDS CORP.                          0.56  
8352         LUXCO INC                                    0.24  
10000        MAJESTIC FINE WINES                          0.25  
3089         SIDNEY FRANK IMPORTING CO                    0.35  
3924         HEAVEN HILL DISTILLERIES                     0.24  
8664         STOLI GROUP,(USA) LLC                        0.36  
9625         WESTERN SPIRITS BEVERAGE CO                  0.21  
2242         DELICATO VINEYARDS INC                       0.16  
1273         CALEDONIA SPIRITS INC                        0.19  
4848         LAIRD & CO                                   0.10  
2561         EDRINGTON AMERICAS                           0.16  
98450        Serralles Us

Compared to the findings in part C, most of these freight-to-dollar percentages are much higher. The highest freight-to-dollar % in part C is 0.54, while the highest in part E is 5.48.

# Deliverable 5

In [23]:
sales = pd.read_csv('salesDecember2016Sample.csv')
sales.head(1)

InventoryId  Store  Brand                   Description   Size  \
0  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL   

   SalesQuantity  SalesDollars  SalesPrice   SalesDate  Volume  \
0              5        114.95       22.99  2016-12-01  750.00   

   Classification  ExciseTax  VendorNo                VendorName  
0               1       3.94      3960  DIAGEO NORTH AMERICA INC

In [24]:
invoices = pd.read_csv('2017PurchasePricesDecSample.csv')
invoices.head(1)

Brand                  Description  Price   Size Volume  Classification  \
0     58  Gekkeikan Black & Gold Sake  12.99  750mL    750               1   

   PurchasePrice  VendorNumber                   VendorName  
0           9.28          8320  SHAW ROSS INT L IMP LTD

In [25]:
Sales = sales[['Description','SalesQuantity','SalesDollars']]
Sales = Sales.groupby(['Description']).sum()
Sales.head(1)

SalesQuantity  SalesDollars
Description                             
(RI) 1                  34      1,195.66

In [26]:
Invoices = invoices[['Description','PurchasePrice']]
Invoices = Invoices.groupby(['Description']).sum()
Invoices.head(1)

PurchasePrice
Description               
(RI) 1               26.92

In [27]:
SalesInvoices = pd.merge(left=Sales, right=Invoices, how='outer', on=['Description'])
SalesInvoices.head(1)

SalesQuantity  SalesDollars  PurchasePrice
Description                                            
(RI) 1               34.00      1,195.66          26.92

In [28]:
SalesInvoices['AverageSalesPrice'] = (SalesInvoices['SalesDollars'] / SalesInvoices['SalesQuantity'])
SalesInvoices.head(1)

SalesQuantity  SalesDollars  PurchasePrice  AverageSalesPrice
Description                                                               
(RI) 1               34.00      1,195.66          26.92              35.17

In [29]:
SalesInvoices['PurchasesDollars'] = (SalesInvoices['PurchasePrice'] * SalesInvoices['SalesQuantity'])
SalesInvoices.head(1)

SalesQuantity  SalesDollars  PurchasePrice  AverageSalesPrice  \
Description                                                                  
(RI) 1               34.00      1,195.66          26.92              35.17   

             PurchasesDollars  
Description                    
(RI) 1                 915.28

In [30]:
SalesInvoices['CMperUnit'] = (SalesInvoices['AverageSalesPrice'] - SalesInvoices['PurchasePrice'])
SalesInvoices.head(1)

SalesQuantity  SalesDollars  PurchasePrice  AverageSalesPrice  \
Description                                                                  
(RI) 1               34.00      1,195.66          26.92              35.17   

             PurchasesDollars  CMperUnit  
Description                               
(RI) 1                 915.28       8.25

In [34]:
SalesInvoices['TotalCM'] = (SalesInvoices['SalesDollars'] - SalesInvoices['PurchasesDollars'])
SalesInvoices.head(1)

SalesQuantity  SalesDollars  PurchasePrice  AverageSalesPrice  \
Description                                                                  
(RI) 1               34.00      1,195.66          26.92              35.17   

             PurchasesDollars  CMperUnit  TotalCM  
Description                                        
(RI) 1                 915.28       8.25   280.38

In [32]:
SalesInvoicesTotal = SalesInvoices.sort_values(by='TotalCM', ascending=False)
SalesInvoicesTotal.head()

SalesQuantity  SalesDollars  PurchasePrice  \
Description                                                                
Kendall Jackson Chard Vt RSV      16,922.00    169,050.78           6.77   
Dom Perignon                         862.00    120,671.38          87.09   
Josh Cellars Cab Svgn Sonoma       9,810.00    128,057.90           8.44   
Smirnoff Traveler                 10,869.00    184,664.31          12.94   
Baileys Irish Cream w/4 Glss       5,430.00    108,545.70          14.50   

                              AverageSalesPrice  PurchasesDollars  CMperUnit  \
Description                                                                    
Kendall Jackson Chard Vt RSV               9.99        114,561.94       3.22   
Dom Perignon                             139.99         75,071.58      52.90   
Josh Cellars Cab Svgn Sonoma              13.05         82,796.40       4.61   
Smirnoff Traveler                         16.99        140,644.86       4.05   
Baileys Irish Cream w/4 Glss              19.99         78,735.00       5.49   

                               TotalCM  
Description                             
Kendall Jackson Chard Vt RSV 54,488.84  
Dom Perignon                 45,599.80  
Josh Cellars Cab Svgn Sonoma 45,261.50  
Smirnoff Traveler            44,019.45  
Baileys Irish Cream w/4 Glss 29,810.70

In [33]:
SalesInvoicesUnit = SalesInvoices.sort_values(by='CMperUnit', ascending=False)
SalesInvoicesUnit.head()

SalesQuantity  SalesDollars  PurchasePrice  \
Description                                                                
Remy Martin Louis XIII                 1.00      3,049.99       2,290.07   
Ch Lafite Rothschild 10 Brdx           1.00      1,599.99       1,103.44   
Ladyburn 41 Yr Old Scotch              1.00      1,999.99       1,550.38   
Ch La Mission Haut-Brion 10            1.00      1,099.99         753.42   
Laphroaig 32 Yr Old Scotch             2.00      2,399.98         875.91   

                              AverageSalesPrice  PurchasesDollars  CMperUnit  \
Description                                                                    
Remy Martin Louis XIII                 3,049.99          2,290.07     759.92   
Ch Lafite Rothschild 10 Brdx           1,599.99          1,103.44     496.55   
Ladyburn 41 Yr Old Scotch              1,999.99          1,550.38     449.61   
Ch La Mission Haut-Brion 10            1,099.99            753.42     346.57   
Laphroaig 32 Yr Old Scotch             1,199.99          1,751.82     324.08   

                              TotalCM  
Description                            
Remy Martin Louis XIII         759.92  
Ch Lafite Rothschild 10 Brdx   496.55  
Ladyburn 41 Yr Old Scotch      449.61  
Ch La Mission Haut-Brion 10    346.57  
Laphroaig 32 Yr Old Scotch     648.16

I did an analysis on contribution margin by product. The Kendall Jackson Chard Vt RSV had the highest total contribution margin in December, as $54,488.84. The product with the highest contribution margin per unit is the Remy Martin Louis XIII.